Adaptive cards

kwinto opened the pull request:
This PR adds support for Adaptive Cards.

Also:
- Improves demo page
- Moves demo page to test to keep the code only relevant for the package in src

To test:

`npm run dev`

`press 'o' + <Enter>`

Note: there seem to be an issue related to Swiper+Adaptive Cards dependencies, it currently breaks test runs.

fabclj commented:
I like the new demo page, but it would be great to integrate all messages types.
Also it would be great to solve the issue about swiper and adaptive card.

fabclj commented:
I have run the adaptive card on webchat locally. I list here the first findings:

![Screenshot from 2024-01-11 12-11-03](https://github.com/Cognigy/chat-components/assets/117462879/09dd47df-718d-4e4a-9036-07ded785bed2)


- Primary button should have height 44px and cursor: pointer
- The card width should be max 295px like all messages
- Textarea scrollbar is not styled as Figma requirments

fabclj commented on src/messages/AdaptiveCards/index.tsx:
```diff
@@ -0,0 +1,110 @@
+
+	const onExecuteAction = useCallback(
+		(action: Action) => {
+			console.log("onExecuteAction", action);
```
please remove debug rest

kwinto replied:
fixed

fabclj commented on src/messages/AdaptiveCards/index.tsx:
```diff
@@ -0,0 +1,110 @@
+	const { message, action: onSendMessage, config } = useMessageContext();
+
+	const getCardPayload = (message: IMessage) => {
+		//@ts-ignore
```
we should refine the TS declarations at some point.

kwinto replied:
This is not an intentional error. Like the one which is explicitly false, which you would use in e.g. test case :)

We should do a next round of interfaces improvements at some point.

fabclj commented on package.json:
```diff
@@ -25,35 +25,43 @@
 		"react": "^17.x || 18.x",
 		"react-dom": "^17.x || 18.x"
+	},
+	"overrides": {
```
this overrides should be also defined on the consumer apps, for example in webchat v3

kwinto replied:
We will address this in later PRs.

kwinto commented:
> * Primary button should have height 44px and cursor: pointer
Fixed

> * The card width should be max 295px like all messages
Full width for AC messages is specified in Figma

> * Textarea scrollbar is not styled as Figma requirments
I fixed it as far as it is possible with CSS. We lack the possibility to have a margin and coloured "track" at the same time on a scrollbar. 

Further improvements here will need a custom AC render or some kind of field override, as we are at the limit of what is possible with to do with pure CSS.

kwinto commented:
> I like the new demo page, but it would be great to integrate all messages types. Also it would be great to solve the issue about swiper and adaptive card.

Tracked in new sotries for the next sprint.

fabclj commented on src/messages/AdaptiveCards/styles.module.css:
```diff
@@ -0,0 +1,189 @@
+	border-color: var(--primary-color);
+}
+
+.adaptivecard-wrapper button {
```
I think we are also missing the button styles when is disabled

kwinto replied:
fixed

fabclj commented:
Text area now has scrollbar but is still not fitting the design requirements

![Screenshot from 2024-01-11 16-20-25](https://github.com/Cognigy/chat-components/assets/117462879/57dcc3d3-8215-4fc1-a372-bd4e09a59fa1)
